File: Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/componentmodel/exceptions/NotNullOrEmptyRequiredException.java
package com.etiennebaudoux.clipboardzanager.componentmodel.exceptions;

public class NotNullOrEmptyRequiredException extends RuntimeException {
    public NotNullOrEmptyRequiredException(String parameterName) {
        super("The value must not be null or empty : " + parameterName);
    }
}

File: ClipboardZanager/Sources/ClipboardZanager.Shared/Exceptions/NotNullOrEmptyRequiredException.cs
using System;

namespace ClipboardZanager.Shared.Exceptions
{
    public sealed class NotNullOrEmptyRequiredException : ArgumentException
    {
        public NotNullOrEmptyRequiredException(string parameterName)
            : base(parameterName, "The value must not be null or empty")
        {
        }
    }
}
